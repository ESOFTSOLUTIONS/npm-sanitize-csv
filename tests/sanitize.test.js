const { sanitizeValues } = require('../index');
const fs = require('fs');

describe('should return errors when arguments are missing', () => {
  test('throws error when delimiter is not provided', () => {
    expect(() => {
      // read csv file
      const csv = fs.readFileSync(__dirname + '/examples/example1.csv', 'utf-8');
      // call function
      sanitizeValues(csv, { quotes: `"` });
    }).toThrow(SyntaxError)
  })

  test('throws error when quotes is not provided', () => {
    expect(() => {
      // read csv file
      const csv = fs.readFileSync(__dirname + '/examples/example1.csv', 'utf-8');
      // call function
      sanitizeValues(csv, { delimiter: ',' });
    }).toThrow(SyntaxError)
  })
})

describe('should return sanitized csv data', () => {
  test('sanitizes csv example 1', () => {
    // read csv file
    const csv = fs.readFileSync(__dirname + '/examples/example1.csv', 'utf-8');
    // call function
    const sanitizationResult = sanitizeValues(csv, { delimiter: ',', quotes: `"` })
    console.log(sanitizationResult, '>> results <<');
    expect(sanitizationResult).toMatchSnapshot()
  })

  test('sanitizes csv example 2 with special characters', () => {
    // read csv file
    const csv = fs.readFileSync(__dirname + '/examples/example2.csv', 'utf-8');
    // call function
    const sanitizationResult = sanitizeValues(csv, { delimiter: ',', quotes: `"` })
    console.log(sanitizationResult, '>> results <<');
    expect(sanitizationResult).toMatchSnapshot()
  })
})

