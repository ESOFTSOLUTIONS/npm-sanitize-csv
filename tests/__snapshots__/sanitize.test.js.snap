// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should return sanitized csv data sanitizes csv example 2 with special characters 1`] = `
"date created,invoice title,invoice number,invoice currency,reference id,customer name,customer email,customer id,invoice URL,status,date paid,item name,quantity,unit price,amount,discount,total invoice amount,paid by bank,stripe receipt,transaction hash,verification type,source,notes,next_invoice,last_invoice,active
2023/01/25,coca cola,'='40,USD,,Eledi,edyrkaj@gmail.com,,https://gilded-dev.firebaseapp.com/invoices/share/ZeWHbUK7UAU8ywSRRDSR,unpaid,,coca cola,1,9,9,0,9,TRUE,,,automatic,invoice,,,,
2023/01/25,cherry coke,'@'41,USD,,coca cola,myrtezaera@gmail.com,,https://gilded-dev.firebaseapp.com/invoices/share/E70zBgV1npZs3gFwUY7u,unpaid,,cherry coke :),10,2.5,25,0,25,TRUE,,,automatic,invoice,,,,
2023/01/25,cola,42,USDC,,coca cola,myrtezaera@gmail.com,,https://gilded-dev.firebaseapp.com/invoices/share/6gawfGTKFbuZZYaH9vmy,unpaid,,cherry,10,2,20,0,20,FALSE,FALSE,,automatic,invoice,,,,"
`;
